lib/dealership.rb

```diff
@@ -0,0 +1,62 @@
+class Dealership
+  attr_reader :inventory
+
+  def initialize(name, address)
+    @name = name
+    @address = address
+    @inventory = []
+  end
+
+  def inventory_count
+    @inventory.size
+  end
+
+  def add_car(car)
+    @inventory << car
+  end
+
+  def has_inventory?
+    @inventory != []
+  end
+
+  def cars_by_make(car_make)
+    @inventory.select do |car|
+      car_make == car.make
+    end
+  end
+
+  def total_value
+    total_cost_of_cars = 0
+    @inventory.map do |car|
+      total_cost_of_cars += car.total_cost
+    end
+    total_cost_of_cars
+  end
+
+  def cars_sorted_by_price
+    sorted_cars = @inventory.sort_by do |car|
+      car.total_cost
+    end
+    sorted_cars
+  end
+
+  def average_price_of_car
+    total_value / @inventory.count
```

Comment:
I might replace @inventory.count with `inventory_count` since we have that method available